src/field_theory/krull_topology.lean

```diff
@@ -287,3 +287,222 @@ begin
+    rw [← p_E_def, ← σ_E_def] at hσ_E,
+    have hσ_F := alg_hom_of_finite_dimensional_of_ultrafilter_spec hF f,
+    rw [← p_F_def, ← σ_F_def] at hσ_F,
+    set res : (F →ₐ[K] L) → (E →ₐ[K] L) := (λ ϕ, ϕ.comp (subalgebra.inclusion hEF))
+      with res_def,
+    have h_pF_pE_res : res ∘ p_F = p_E := rfl,
+    have h_maps_commute : ((f.map p_F).map res : filter (E →ₐ[K] L)) = f.map p_E,
+    { rw [ultrafilter.map_map, h_pF_pE_res] },
+    have hEf := alg_hom_of_finite_dimensional_of_ultrafilter_spec hE f,
+    rw [← σ_E_def, ← p_E_def] at hEf,
+    have hFf := alg_hom_of_finite_dimensional_of_ultrafilter_spec hF f,
+    rw [← σ_F_def, ← p_F_def] at hFf,
+    have hFf' : (ultrafilter.map p_F f) = (pure σ_F : ultrafilter (F →ₐ[K] L)) :=
+    ultrafilter.coe_inj.mp hFf,
+    rw [hEf, hFf', ultrafilter.map_pure] at h_maps_commute,
+    have h := filter.pure_injective h_maps_commute,
+    rw res_def at h,
+    exact h.symm,
+  end
+
+/-- For each `x : L`, the intermediate field `K(x)` is finite dimensional over `K`, so
+  `alg_hom_of_finite_dimensional_of_ultrafilter` gives a map `ϕₓ : K(x) → L`. We define
+  `function_of_ultrafilter h_int f` to be the function taking `x` to `ϕₓ(x)` for all `x : L`. -/
+noncomputable def function_of_ultrafilter {K L : Type*} [field K] [field L] [algebra K L]
+  (h_int : algebra.is_integral K L) (f : ultrafilter (L ≃ₐ[K] L)) :
+  (L → L) :=
+λ x, (alg_hom_of_finite_dimensional_of_ultrafilter
+(intermediate_field.adjoin.finite_dimensional (h_int x)) f)
+(⟨x, intermediate_field.mem_adjoin_simple_self K x⟩)
+
+
+lemma function_of_ultrafilter_spec {K L : Type*} [field K] [field L] [algebra K L]
+  (h_int : algebra.is_integral K L) (f : ultrafilter (L ≃ₐ[K] L)) {E : intermediate_field K L}
+  (hE : finite_dimensional K E) (x : E) :
+  (function_of_ultrafilter h_int f) x = (alg_hom_of_finite_dimensional_of_ultrafilter hE f) x :=
+begin
+   have h_le : intermediate_field.adjoin K {(x : L)} ≤ E,
+   { apply intermediate_field.gc.l_le,
+     simp only [set_like.coe_mem, set_like.mem_coe, set.singleton_subset_iff, set.le_eq_subset] },
+   have h_Kx : finite_dimensional K (intermediate_field.adjoin K {(x : L)}) :=
+   intermediate_field.adjoin.finite_dimensional (h_int x),
+   let h_functor := alg_hom_of_finite_dimensional_of_ultrafilter_functor h_Kx f hE h_le,
+   have h : (function_of_ultrafilter h_int f) x =
+   (alg_hom_of_finite_dimensional_of_ultrafilter h_Kx f) ⟨x,
+   intermediate_field.mem_adjoin_simple_self K x⟩ := rfl,
+   rw [h, h_functor],
+   let x_in_Kx : intermediate_field.adjoin K {(x : L)} := ⟨(x : L),
+   intermediate_field.mem_adjoin_simple_self K (x : L)⟩,
+   have h_inclusion_mk : (subalgebra.inclusion h_le) ⟨x, intermediate_field.mem_adjoin_simple_self K
+   (x : L)⟩ =
+   ⟨x, h_le (intermediate_field.mem_adjoin_simple_self K (x : L))⟩ := rfl,
+   simp [h_inclusion_mk],
+end
+
+lemma adj_finset_finite_dimensional {K L : Type*} [field K] [field L] [algebra K L]
```

Comment:
This lemma has nothing to do here. I guess it belongs in `field_theory.intermediate_field`?

Reply:
I have moved it to field_theory.adjoin